### DISCUSSION TITLE
How do I draw a grid of dots?

### DISCUSSION THREAD

**Discussion by dbuzinin** [pixijs/pixijs#9021] [Q&A]
I use pixi-viewport to pan and zoom. I need to draw a background consisting of dots every 10px. When panning and zooming, the dots need to be redrawn so that they always fill the entire screen.

I've tried doing Spite from a white texture and drawing dots with Graphics, but both options affect performance.
`
drawGrid(): void {
    const { x, y } = this._viewport.toWorld(0, 0);
    const horizontal = Math.round(this._viewport.worldScreenWidth / 10)
    const vertical = Math.round(this._viewport.worldScreenHeight / 10)
    this._grid.clear();
    this._grid.lineStyle({
      width: 2,
      color: 0xbbc0c3,
    });
    for (let i = 0; i < vertical; i++) {
      for (let j = 0; j < horizontal; j++) {
        this._grid.moveTo(x + j * 10, y + i * 10)
        this._grid.lineTo(x + j * 10 + 1, y + i * 10)
      }
    }
  }
`

Is there any way to draw the grid quickly?
![image](https://user-images.githubusercontent.com/75254600/209561979-e119d9b8-a890-4bcd-a4d5-dad0f7c60028.png)

**Comment by bigtimebuddy** [comment]
Using TilingSprite instead?

**Reply by dbuzinin** [reply]
Thank you! In terms of performance, it works fine. But when scaling, I get strange effects.
For example, when setting tileScale = 0.12

I made a working example, try spinning the mouse wheel - https://codesandbox.io/s/pixi-playground-forked-tp617c

![image](https://user-images.githubusercontent.com/75254600/209689027-148013ba-4168-4d92-b15e-a3e518550ef8.png)
